integration-tests/tests/common/mod.rs

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! This crate contains the common code for integration tests
+
+#![deny(clippy::all)]
+#![deny(missing_docs)]
+
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::time::Duration;
+use wait_timeout::ChildExt;
+
+#[cfg_attr(not(any(has_sgx, has_sev)), ignore)]
+pub fn keep_under_test() -> &'static str {
+    let mut result = "";
+    if cfg!(has_sgx) {
+        result = "enarx-keep-sgx";
+    }
+    if cfg!(has_sev) {
+        result = "enarx-keep-sev";
+    }
+    result
+}
+
+pub struct IntegrationTest {
+    pub wksp_root: PathBuf,
+    pub keep: PathBuf,
+    pub payload: PathBuf,
+    pub shim: PathBuf,
+}
+
+impl IntegrationTest {
+    pub fn new(bin: &str) -> Self {
+        let enarx_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let enarx_dir = enarx_dir.parent().unwrap();
+        let keep_type = keep_under_test();
+        let local_wksp_root = enarx_dir.to_path_buf();
+        let local_keep = find_filepath(
+            enarx_dir
+                .join(format!("{}/target/", keep_type))
+                .to_path_buf(),
+            format!("{}", keep_type).as_str(),
+        )
+        .unwrap();
+        let local_payload = find_filepath(
+            enarx_dir
+                .join("integration-tests/target/x86_64-unknown-linux-musl/")
+                .to_path_buf(),
+            bin,
+        )
+        .unwrap();
+        let local_shim = find_filepath(
+            enarx_dir
+                .join(format!(
+                    "{}-shim/target/x86_64-unknown-linux-musl/",
+                    keep_type
+                ))
+                .to_path_buf(),
+            format!("{}-shim", keep_type).as_str(),
+        )
+        .unwrap();
+        Self {
+            wksp_root: local_wksp_root,
+            keep: local_keep,
+            payload: local_payload,
+            shim: local_shim,
+        }
+    }
+
+    pub fn run_test(self: Self, timeout: usize, exit_status: i32) -> () {
```

Comment:
```suggestion
    pub fn run(self: Self, timeout: usize, exit_status: i32) -> () {
```

Since the type is `IntegrationTest` we know it is a test. Hence `run_test()` is redundant.